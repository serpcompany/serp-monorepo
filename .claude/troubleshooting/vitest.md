https://vitest.dev/guide/common-errors.html