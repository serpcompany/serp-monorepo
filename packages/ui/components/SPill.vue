<script setup lang="ts">
  import type { Category } from '@serp/types/types'

  defineProps<{
    items: Category[]
    baseSlug: string
  }>()
</script>

<template>
  <div class="flex flex-wrap gap-2">
    <NuxtLink
      v-for="item in items"
      :key="item.id"
      :to="`/${baseSlug}/${item.slug}/`"
      class="rounded-full border border-blue-300 bg-blue-50 px-3 py-1 text-sm font-medium text-blue-600 hover:bg-blue-100 dark:border-blue-700 dark:bg-blue-900/30 dark:text-blue-300 dark:hover:bg-blue-800/50"
    >
      {{ item.name }}
    </NuxtLink>
  </div>
</template>
