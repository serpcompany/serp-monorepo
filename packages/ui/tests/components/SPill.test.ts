import { describe, expect, it } from 'vitest'
import SPill from '../../components/SPill.vue'
import ComponentRender from '../componentRender'

describe('sPill', () => {
  const baseSlug = 'category'

  it.each([
    ['with no items', { props: { items: [], baseSlug } }],
    [
      'with one item',
      {
        props: { items: [{ id: 1, slug: 'item-1', name: 'Item 1' }], baseSlug },
      },
    ],
    [
      'with multiple items',
      {
        props: {
          items: [
            { id: 1, slug: 'item-1', name: 'Item 1' },
            { id: 2, slug: 'item-2', name: 'Item 2' },
            { id: 3, slug: 'item-3', name: 'Item 3' },
          ],
          baseSlug,
        },
      },
    ],
  ])(
    'renders %s correctly',
    async (desc: string, options: { props: unknown }) => {
      const html = await ComponentRender(`SPill ${desc}`, options, SPill)
      expect(html).toMatchSnapshot()
    },
  )
})
