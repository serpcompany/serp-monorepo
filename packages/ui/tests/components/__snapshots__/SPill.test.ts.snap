// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPill > renders with multiple items correctly 1`] = `
<div class="
  flex
  flex-wrap
  gap-2
">
  <a
    class="
      bg-blue-50
      border
      border-blue-300
      dark:bg-blue-900/30
      dark:border-blue-700
      dark:hover:bg-blue-800/50
      dark:text-blue-300
      font-medium
      hover:bg-blue-100
      px-3
      py-1
      rounded-full
      text-blue-600
      text-sm
    "
    href="/category/item-1/"
  >Item 1</a>
  <a
    class="
      bg-blue-50
      border
      border-blue-300
      dark:bg-blue-900/30
      dark:border-blue-700
      dark:hover:bg-blue-800/50
      dark:text-blue-300
      font-medium
      hover:bg-blue-100
      px-3
      py-1
      rounded-full
      text-blue-600
      text-sm
    "
    href="/category/item-2/"
  >Item 2</a>
  <a
    class="
      bg-blue-50
      border
      border-blue-300
      dark:bg-blue-900/30
      dark:border-blue-700
      dark:hover:bg-blue-800/50
      dark:text-blue-300
      font-medium
      hover:bg-blue-100
      px-3
      py-1
      rounded-full
      text-blue-600
      text-sm
    "
    href="/category/item-3/"
  >Item 3</a>
</div>
`;

exports[`SPill > renders with no items correctly 1`] = `
<div class="
  flex
  flex-wrap
  gap-2
"></div>
`;

exports[`SPill > renders with one item correctly 1`] = `
<div class="
  flex
  flex-wrap
  gap-2
">
  <a
    class="
      bg-blue-50
      border
      border-blue-300
      dark:bg-blue-900/30
      dark:border-blue-700
      dark:hover:bg-blue-800/50
      dark:text-blue-300
      font-medium
      hover:bg-blue-100
      px-3
      py-1
      rounded-full
      text-blue-600
      text-sm
    "
    href="/category/item-1/"
  >Item 1</a>
</div>
`;
